examples/tester.rs

```diff
@@ -0,0 +1,309 @@
+
+                                        let xpart = x.clamp(0.0, frame_w as f32 - 1.0) as usize;
+                                        let ypart =
+                                            y.clamp(0.0, frame_h as f32 - 1.0) as usize * stride;
+                                        frame[ypart + xpart] = 0xffffffff;
+                                    }
+                                }
+                            }
+                        };
+
+                    // Animation.
+                    let x = 25.0;
+                    let y = 25.0;
+
+                    static START: std::sync::OnceLock<web_time::Instant> =
+                        std::sync::OnceLock::new();
+                    let start = START.get_or_init(web_time::Instant::now);
+                    let time = web_time::Instant::now().duration_since(*start).as_secs_f32();
+
+                    let xo = (time).sin() * 25.0;
+                    let yo = (time).cos() * 25.0;
+                    draw_line(x, y, xo, yo, 1);
+                    draw_line(x, y, -xo, -yo, 1);
+
+                    // Any new lines to draw from input?
+                    let diameter = self.has_pressure.map(|x| x * 10.0).unwrap_or(0.0).ceil();
+                    if self.drawing {
+                        draw_line(
+                            self.old_posx,
+                            self.old_posy,
+                            self.posx - self.old_posx,
+                            self.posy - self.old_posy,
+                            diameter as u32 + 1,
+                        );
+                    }
+                    let radius = (diameter * 0.5).ceil();
+
+                    let x = (self.posx.min(self.old_posx) - radius).max(0.0);
+                    let y = (self.posy.min(self.old_posy) - radius).max(0.0);
+                    let w =
+                        (self.posx.max(self.old_posx) - x + 1.0 + radius).min(frame_w as f32 - x);
+                    let h =
+                        (self.posy.max(self.old_posy) - y + 1.0 + radius).min(frame_h as f32 - y);
+
+                    *rects_ref =
+                        vec![[0, 0, 50, 50], rect1, [x as u32, y as u32, w as u32, h as u32]];
+                    let mut damaged = rects_ref.clone();
+                    damaged.extend_from_slice(&self.last_draw);
+                    damaged
+                });
+
+                self.old_posx = self.posx;
+                self.old_posy = self.posy;
+
+                self.last_draw = rects;
+
+                // For contiguous redraw loop you can request a redraw from here.
+                window.request_redraw();
+                // Don't run hundreds of thousands of times per second even if the platform asks.
+                // (Can't sleep on web, so gated off there. Browsers won't ask for too much.)
+                #[cfg(not(web_platform))]
+                {
+                    std::thread::sleep(std::time::Duration::from_millis(1));
+                }
```

Comment:
The fact that you currently need this kind of stuff is _exactly_ why this example is so useful!